### ISSUE TITLE
can doxygen deal with function generate by Factory

### ISSUE THREAD

**Issue by hfutyezi** [doxygen/doxygen#6428]
`            $topicBusiness = Factory::create('TopicBusiness');
            $data = $topicBusiness->getEntryData($pageId,$tagId,$styleId,$topicNum,$warehouse,$client,$isRank);`
When i read html ,I found **$topicBusiness->getEntryData** not exist ，is it not support？

**Comment by albert-github** [comment]
Can you please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem?

**Comment by albert-github** [comment]
You wrote "When i read html ,I found $topicBusiness->getEntryData not exist ，is it not support？" can you (as I don't know where to look for the eerror):
- point out the page where you saw this problem (picture)
- when necessary add your resulting html tree as zipped attachment to this issue?

**Comment by hfutyezi** [comment]
[demo.zip](https://github.com/doxygen/doxygen/files/2262018/demo.zip)
Add picture

**Comment by albert-github** [comment]
I can see the problem in the small project as well (nicely cut down from your big project).

**Comment by hfutyezi** [comment]
this problem is not common,Maybe in other project it is not write as this format 
eg:
$model = Loader::model('TemplateDetail');

this is another format I saw

**Comment by hfutyezi** [comment]
Maybe we need a new configure ? Congfigure each PHP project Factory format,I think

**Comment by albert-github** [comment]
I don't get the intention of the last 2 comments.

I've seen in the example that `$topicBusiness = Factory::create('TopicBusiness');` is giving the reference in the graph. Looks like similar constructs in Cpp do work.

**Comment by hfutyezi** [comment]
We need a common solution about this issue,In my example we get a object with Class TopicBusiness which was created by "Factory Pattern"(  Factory::create('TopicBusiness')  )。Maybe in other PHP project,it is written as 
$topicBusiness = Loader::model('TopicBusiness');  it is just a name.is not common .
-------------
I don't know if you understand it^_^

**Comment by albert-github** [comment]
Problem looks to be that the `$topicBusiness` does not have a (to doxygen known) type. Would it be possible to do something like (not sure about the syntax):
```
            TopicBusiness $topicBusiness ;
            $topicBusiness = Factory::create('TopicBusiness');
            ...
```
 in this case the functions are shown.

**Comment by hfutyezi** [comment]
Yes,In Java or other language it maybe ok,In Php,It's also okay when we don't declare TopicBusiness

**Comment by albert-github** [comment]
I know that php doesn't need to declare the variable, this results in the fact that doxygen doesn't know the type and cannot find the function as it does not evaluate the results of e.g. the `create` function.

Question was is it feasible todo something like:
```
            TopicBusiness $topicBusiness ;
            $topicBusiness = Factory::create('TopicBusiness');
            ...
```
so explicitly declaring the variable.

**Comment by hfutyezi** [comment]
Hah,In my own Project ,I will do this work so that we can use doxygen do get callGraph,
But my work is to analysis other people's Project ,It maybe in anywhere in his project
It sounds like In these project ,I can't use doxygen to analysis these project,
Anyway ,Thank you very much.
